akka-typed/src/main/scala/akka/typed/internal/DeathWatch.scala

```diff
@@ -70,13 +80,18 @@ private[typed] trait DeathWatch[T] {
    */
   protected def watchedActorTerminated(actor: ARImpl, failure: Throwable): Boolean = {
     removeChild(actor)
-    if (watching.contains(actor)) {
+    watching.get(actor).foreach { optionalMessage ⇒
       maintainAddressTerminatedSubscription(actor) {
         watching -= actor
       }
       if (maySend) {
-        val t = Terminated(actor)(failure)
-        next(Behavior.interpretSignal(behavior, ctx, t), t)
+        optionalMessage match {
+          case None ⇒
+            val t = Terminated(actor)(failure)
+            next(Behavior.interpretSignal(behavior, ctx, t), t)
+          case Some(msg) ⇒
+            next(Behavior.interpretMessage(behavior, ctx, msg), msg)
```

Comment:
I can't see that this is implemented in akka/actor/dungeon/DeathWatch.scala.

Add another test that is running with both native and adapted system, something like I did in https://github.com/akka/akka/pull/22765/files#diff-515cfeed32846d4013a3da4829385d84R65